prow/cmd/sinker/main.go

```diff
@@ -124,48 +124,35 @@ func (c *controller) clean() {
 		return
 	}
 
-	// Only delete pod if its prowjob is marked as finished
-	isFinished := make(map[string]bool)
-
-	maxProwJobAge := c.configAgent.Config().Sinker.MaxProwJobAge
-	for _, prowJob := range prowJobs {
-		// Handle periodics separately.
-		if prowJob.Spec.Type == kube.PeriodicJob {
-			continue
-		}
-		if !prowJob.Complete() {
-			continue
-		}
-		isFinished[prowJob.ObjectMeta.Name] = true
-		if time.Since(prowJob.Status.StartTime.Time) <= maxProwJobAge {
-			continue
-		}
-		if err := c.kc.DeleteProwJob(prowJob.ObjectMeta.Name); err == nil {
-			c.logger.WithFields(pjutil.ProwJobFields(&prowJob)).Info("Deleted prowjob.")
-		} else {
-			c.logger.WithFields(pjutil.ProwJobFields(&prowJob)).WithError(err).Error("Error deleting prowjob.")
-		}
-	}
-
 	// Keep track of what periodic jobs are in the config so we will
 	// not clean up their last prowjob.
+	// This allows horologium to determine time since the last run for periodics.
 	isActivePeriodic := make(map[string]bool)
 	for _, p := range c.configAgent.Config().Periodics {
 		isActivePeriodic[p.Name] = true
 	}
-
-	// Get the jobs that we need to retain so horologium can continue working
-	// as intended.
 	latestPeriodics := pjutil.GetLatestProwJobs(prowJobs, kube.PeriodicJob)
-	for _, prowJob := range prowJobs {
-		if prowJob.Spec.Type != kube.PeriodicJob {
-			continue
-		}
 
-		latestPJ := latestPeriodics[prowJob.Spec.Job]
-		if isActivePeriodic[prowJob.Spec.Job] && prowJob.ObjectMeta.Name == latestPJ.ObjectMeta.Name {
-			// Ignore deleting this one.
-			continue
+	// Only delete pod if either
+	// - The prowjob is missing
+	// - The prowjob is marked as finished AND either
+	//   - The pod is completed.
+	//   - The pod reaches the max age.
+	// Note that this intentionally allows pods to continue running if the PJ is
+	// aborted, but the pod is not explicitly deleted.
+	isFinished := make(map[string]bool)
+	existing := make(map[string]bool)
+
+	maxProwJobAge := c.configAgent.Config().Sinker.MaxProwJobAge
+	for _, prowJob := range prowJobs {
+		existing[prowJob.ObjectMeta.Name] = true
```

Comment:
`ObjectMetaData` is an embedded field so I believe you can just do `prowJob.Name`